rcl/src/rcl/arguments.c

```diff
@@ -0,0 +1,403 @@
+      return RCL_RET_ERROR;
+    }
+    if (RMW_NODE_NAME_VALID != validation_result) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        allocator,
+        "node name prefix invalid: %s", rmw_node_name_validation_result_string(validation_result));
+      return RCL_RET_INVALID_REMAP_RULE;
+    }
+  }
+
+  // Figure out what type of rule this is, default is to apply to topic and service names
+  rcl_remap_type_t type = RCL_TOPIC_REMAP | RCL_SERVICE_REMAP;
+  if (0 == strncmp("__ns", match_begin, len_match)) {
+    type = RCL_NAMESPACE_REMAP;
+  } else if (0 == strncmp("__node", match_begin, len_match)) {
+    type = RCL_NODENAME_REMAP;
+  }
+
+  if (type & (RCL_TOPIC_REMAP | RCL_SERVICE_REMAP)) {
+    // Replacement must be a valid topic name
+    int validation_result;
+    size_t invalid_index;
+    rcl_ret_t ret = rcl_validate_topic_name(replacement_begin, &validation_result, &invalid_index);
+    if (ret != RCL_RET_OK) {
+      return RCL_RET_ERROR;
+    } else if (validation_result != RCL_TOPIC_NAME_VALID) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        allocator,
+        "replacement is invalid: %s", rcl_topic_name_validation_result_string(validation_result));
+      return RCL_RET_INVALID_REMAP_RULE;
+    }
+    // Match must be a valid topic name
+    ret = rcl_validate_topic_name_with_size(
+      match_begin, len_match, &validation_result, &invalid_index);
+    if (ret != RCL_RET_OK) {
+      return RCL_RET_ERROR;
+    } else if (validation_result != RCL_TOPIC_NAME_VALID) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        allocator,
+        "match is invalid: %s", rcl_topic_name_validation_result_string(validation_result));
+      return RCL_RET_INVALID_REMAP_RULE;
+    }
+  } else if (RCL_NAMESPACE_REMAP == type) {
+    int validation_result;
+    size_t invalid_idx;
+    if (RMW_RET_OK != rmw_validate_namespace(replacement_begin, &validation_result, &invalid_idx)) {
+      RCL_SET_ERROR_MSG("failed to run check on namespace", allocator);
+      return RCL_RET_ERROR;
+    }
+    if (RMW_NAMESPACE_VALID != validation_result) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        allocator,
+        "namespace is invalid: %s", rmw_namespace_validation_result_string(validation_result));
+      return RCL_RET_INVALID_REMAP_RULE;
+    }
+  } else if (RCL_NODENAME_REMAP == type) {
+    int validation_result;
+    size_t invalid_idx;
+    if (RMW_RET_OK != rmw_validate_node_name(replacement_begin, &validation_result, &invalid_idx)) {
+      RCL_SET_ERROR_MSG("failed to run check on node name", allocator);
+      return RCL_RET_ERROR;
+    }
+    if (RMW_NODE_NAME_VALID != validation_result) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        allocator,
+        "node name is invalid: %s", rmw_node_name_validation_result_string(validation_result));
+      return RCL_RET_INVALID_REMAP_RULE;
+    }
+  }
+
+  // Rule is valid, construct a structure for it
+  output_rule->allocator = allocator;
```

Comment:
Should the function check if the pointer is no `NULL` before dereferencing it?

Reply:
It's probably ok since the function is internal and only used in one place, but it's better have code be defensive so I added checks to this function in 01a13889c9669b9b8dfbd6777f0964601e44a5f8. I didn't find other unchecked arguments in this file. Mind commenting on the other places you saw?